components/brave_news/browser/feed_controller.cc

```diff
@@ -183,29 +163,76 @@ void FeedController::EnsureFeedIsCached(
 }
 
 void FeedController::UpdateIfRemoteChanged(
-    const SubscriptionsSnapshot& subscriptions) {
-  // If already updating, nothing to do,
-  // we don't want to collide with an update
-  // which starts and completes before our HEAD
-  // request completes (which admittedly is very unlikely).
+    const SubscriptionsSnapshot& subscriptions,
+    HashCallback callback) {
+  // If already updating, update with the hash once the update is complete.
+  // We don't want to collide with an update which starts and completes before
+  // our HEAD request completes (which admittedly is very unlikely).
   if (is_update_in_progress_) {
+    on_current_update_complete_->Post(
+        FROM_HERE,
+        base::BindOnce(
+            [](base::WeakPtr<FeedController> controller,
+               HashCallback callback) {
+              if (!controller) {
+                return;
+              }
+
+              std::move(callback).Run(controller->current_feed_.hash);
+            },
+            weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
```

Comment:
Can we dedupe this with lambda inside this function? I think it's okay to capture as reference as it's a short period of time

```
auto post_hash_update_notification = [&](){
          base::BindOnce(
            [](base::WeakPtr<FeedController> controller,
               HashCallback callback) {
              if (!controller) {
                return;
              }

              std::move(callback).Run(controller->current_feed_.hash);
            },
            weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
};


...


  if (is_update_in_progress_) {
    post_hash_update_notification();
```

Reply:
Good call - Wrapped it in a OnceClosure instead :smile: